### ISSUE TITLE
Using nginx-proxy with cloudflare full ssl config

### ISSUE THREAD

**Issue by TheMisir** [nginx-proxy/nginx-proxy#1475]
I'm unable to use cloudflare with nginx-proxy. Because I'm using full ssl config (which requires HTTPS connection between server and cloudflare) there's something that's not working.

Here's simple network graph:

```
                                                                          +----HTTP----> Virtual host 1
                                                                          |
User ------ HTTPS -----> Cloudflare ------ HTTPS ------> nginx-proxy -----|----HTTP----> Virtual host 2
                                                                          |
                                                                          +----HTTP----> Virtual host 3
```

Here's my compose config:

```yml
version: '3.4'

services:
  db:
    image: mdillon/postgis
    volumes:
      - "/var/docker/postgresql/.sock:/var/run/postgres/postgres.sock"
      - "/var/docker/postgresql/data:/var/lib/postgresql/data"
      - "/var/docker/postgresql/initdb:/docker-entrypoint-initdb.d/"

  proxy:
    image: jwilder/nginx-proxy
    volumes:
      - "/var/run/docker.sock:/tmp/docker.sock:ro"
      - "/var/docker/certs:/etc/nginx/certs"
    ports:
      - "80:80"
      - "443:443"

# virtual servers ...
```

Result:

![image](https://user-images.githubusercontent.com/29130291/88076683-6b575680-cb8b-11ea-8398-66f9da3361d9.png)

**Comment by tkw1536** [comment]
I don't know exactly how Cloudflare works, but in order for https certificate generation to succeed, port 80 needs to be accessible from the open web. From the diagram above that doesn't seem to be the case. See https://github.com/nginx-proxy/docker-letsencrypt-nginx-proxy-companion#requirements. 

Also, did you setup `docker-letsencrypt-nginx-proxy-companion`? It's missing from the configuration you posted.

**Comment by TheMisir** [comment]
> Also, did you setup docker-letsencrypt-nginx-proxy-companion? It's missing from the configuration you posted.

No I have not installed "docker-letsencrypt-nginx-proxy-companion".

The issue was solved by itself after a few days. I think the issue is related to cloudflare cache.

**Comment by Xzelsius** [comment]
@TheMisir I am struggling with the same right now.

Can you guide me on how you named your .pem, .key and .cert files?
And what environment vars did you have to set on your containers?

**Comment by TheMisir** [comment]
> @TheMisir I am struggling with the same right now.
> 
> Can you guide me on how you named your .pem, .key and .cert files?
> And what environment vars did you have to set on your containers?

I issued origin certificates for domain.com and *.domain.com then I get only one certificate for root domain and subdomains. Renamed certificate to "domain.com.crt" (not .pem) and key to "domain.com.key" (as described [here](https://github.com/nginx-proxy/nginx-proxy#wildcard-certificates)).

I hope it works. If it doesn't:

What I did next is I'm not sure it really helped or not but I think worth trying.

![image](https://user-images.githubusercontent.com/29130291/90825484-09d10600-e34a-11ea-8eb3-dc24be12a593.png)

I turned on then off the switch labeled "Authenticated Origin Pulls". And website become available after doing that but I'm not sure if it was really helped.
